build.go

```diff
@@ -19,91 +17,55 @@ type DependencyManager interface {
 
-		dependency, err := dependencyManager.Resolve(
-			filepath.Join(context.CNBPath, "buildpack.toml"),
-			"cyclonedx-node-module",
-			"*",
-			context.Stack,
+		var (
+			bom sbom.SBOM
+			err error
 		)
+		duration, err := clock.Measure(func() error {
+			bom, err = sbom.Generate(context.WorkingDir)
+			return err
+		})
 		if err != nil {
 			return packit.BuildResult{}, err
 		}
-		logger.Subprocess("Selected %s version: %s", dependency.Name, dependency.Version)
+
+		logger.Action("Completed in %s", duration.Round(time.Millisecond))
 		logger.Break()
 
-		cycloneDXNodeModuleLayer, err := context.Layers.Get("cyclonedx-node-module")
+		layer, err := context.Layers.Get("node-module-bom")
 		if err != nil {
 			return packit.BuildResult{}, err
 		}
 
-		cachedSHA, ok := cycloneDXNodeModuleLayer.Metadata["dependency-sha"].(string)
-		if ok && cachedSHA == dependency.SHA256 {
-			logger.Process("Reusing cached layer %s", cycloneDXNodeModuleLayer.Path)
-			logger.Break()
-		} else {
-			logger.Process("Executing build process")
-			cycloneDXNodeModuleLayer, err = cycloneDXNodeModuleLayer.Reset()
-			if err != nil {
-				return packit.BuildResult{}, err
-			}
-			logger.Subprocess("Installing %s %s", dependency.Name, dependency.Version)
-			duration, err := clock.Measure(func() error {
-				return dependencyManager.Deliver(dependency, context.CNBPath, cycloneDXNodeModuleLayer.Path, context.Platform.Path)
-			})
-			if err != nil {
-				return packit.BuildResult{}, err
-			}
-
-			logger.Action("Completed in %s", duration.Round(time.Millisecond))
-			logger.Break()
-
-			cycloneDXNodeModuleLayer.Metadata = map[string]interface{}{
-				"dependency-sha": dependency.SHA256,
-				"built_at":       clock.Now().Format(time.RFC3339Nano),
-			}
-		}
-
-		cycloneDXNodeModuleLayer.Cache = true
-
-		logger.Process("Configuring environment")
-		logger.Subprocess("Appending %s onto PATH", dependency.Name)
-		logger.Break()
-
-		os.Setenv("PATH", fmt.Sprint(os.Getenv("PATH"), string(os.PathListSeparator), filepath.Join(cycloneDXNodeModuleLayer.Path, "bin")))
-
-		toolBOM := dependencyManager.GenerateBillOfMaterials(dependency)
-
-		logger.Process("Running %s", dependency.Name)
-		var moduleBOM []packit.BOMEntry
-		duration, err := clock.Measure(func() error {
-			moduleBOM, err = nodeModuleBOM.Generate(context.WorkingDir)
-			return err
-		})
+		layer, err = layer.Reset()
 		if err != nil {
 			return packit.BuildResult{}, err
 		}
 
-		logger.Action("Completed in %s", duration.Round(time.Millisecond))
-		logger.Break()
+		layer.Launch = true
+
+		layer.SBOM.Set("cdx.json", bom.Format(sbom.CycloneDXFormat))
+		layer.SBOM.Set("syft.json", bom.Format(sbom.SyftFormat))
+		layer.SBOM.Set("spdx.json", bom.Format(sbom.SPDXFormat))
+
+		bomEntries := make(packit.SBOMEntries)
+		bomEntries.Set("cdx.json", bom.Format(sbom.CycloneDXFormat))
+		bomEntries.Set("syft.json", bom.Format(sbom.SyftFormat))
+		bomEntries.Set("spdx.json", bom.Format(sbom.SPDXFormat))
```

Comment:
Why do we have this set of BOM entries? Is this not duplicative of what already is attached to the layer?

Reply:
No real reason. We were having unrelated issues with the layer not showing up at launch so we added the entries to the launch metadata too.